packages/quri_parts/jijmodeling_transpiler_quantum/quri_parts/qaoa/to_qaoa_quri.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+import qiskit as qk
+import jijmodeling as jm
+import jijmodeling.transpiler as jmt
+from jijmodeling_transpiler_quantum.core import qubo_to_ising
+from .ising_hamiltonian_quri import to_ising_operator_from_qubo_quri
+from quri_parts.circuit import LinearMappedUnboundParametricQuantumCircuit
+from quri_parts.core.operator import Operator
+from math import pi
+
+
+class QuriQAOAAnsatzBuilder:
+    def __init__(
+        self,
+        pubo_builder: jmt.core.pubo.PuboBuilder,
+        num_vars: int,
+        compiled_instance: jmt.core.CompiledInstance,
+    ):
+        self.pubo_builder = pubo_builder
+        self.num_vars = num_vars
+        self.compiled_instance = compiled_instance
+
+    @property
+    def var_map(self) -> dict[str, tuple[int, ...]]:
+        return self.compiled_instance.var_map.var_map
+
+    def get_hamiltonian(
+        self,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[Operator, float]:
+        qubo, constant = self.pubo_builder.get_qubo_dict(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        ising_operator, ising_const = to_ising_operator_from_qubo_quri(
+            qubo, self.num_vars
+        )
+        return ising_operator, ising_const + constant
+
+    def get_qaoa_ansatz(
+        self,
+        p: int,
+        multipliers: dict = None,
+        detail_parameters: dict = None,
+    ) -> tuple[UnboundParametricQuantumCircuit, operator, float]:
+        ising_operator, constant = self.get_hamiltonian(
+            multipliers=multipliers, detail_parameters=detail_parameters
+        )
+        pauli_terms = []
+        coeff_terms = []
+        pauli_z = []
+        pauli_zz = []
+        pauli_z_coeff = []
+        pauli_zz_coeff = []
+
+        for i in range(ising_operator.n_terms - 1):
+            keys = list(ising_operator.keys())[i]
+            pauli_terms.append(keys.index_and_pauli_id_list[0])
+            items = list(ising_operator.items())[i]
+            coeff_terms.append(items[1])
+
+        quri_QAOAANSATZ = LinearMappedUnboundParametricQuantumCircuit(self.num_vars)
+
+        pauli_z = pauli_terms[: self.num_vars]
+        pauli_zz = pauli_terms[self.num_vars :]
+        pauli_zz = [sorted(sublist) for sublist in pauli_zz]
+        pauli_z_coeff = coeff_terms[: self.num_vars]
+        pauli_zz_coeff = coeff_terms[self.num_vars :]
+
+        for i in pauli_z:
+            quri_QAOAANSATZ.add_H_gate(i[0])
+
+        for i in range(p):
+            Gamma = quri_QAOAANSATZ.add_parameters(f"gamma{i}")
+            Beta = quri_QAOAANSATZ.add_parameters(f"beta{i}")
+
+            for i in zip(pauli_z, pauli_z_coeff):
```

Comment:
DO NOT use the same value name inside of the loop, which you use the outer loop.
Please name the value more correctly.

Reply:
https://github.com/Jij-Inc/JijModelingTranspilerQuantum/blob/b88530ff8d00e00de90e4d17bfd139030df16831/packages/quri_parts/jijmodeling_transpiler_quantum/quri_parts/qaoa/to_qaoa.py#L59-L78
Rearranged the name of value in the loop.